### ISSUE TITLE
Usage update required srcs

### ISSUE THREAD

**Issue by WoolnerWoofieWool** [cyr19/MENLI#2]
Hello. In your usage doco you provide the following:

from menli.MENLI import MENLI

scorer = MENLI(direction="rh", formula="e", nli_weight=0.2, \
                combine_with="MoverScore", model="D", cross_lingual=False)
# refs and hyps in form of list of String
scorer.score_all(refs=refs, hyps=hyps) 

however it appears that we also need to provide a list 'srcs'? It seems to execute if I provide an empty list of srcs and then lists of texts for refs (reference docs?) and hyps (hypotheses?). What is srcs?

If I supply an empty list for srcs I can generate a score and have set combine_with to "None" as I'm interested in the NLI without ensemble.

**Comment by WoolnerWoofieWool** [comment]
I am getting some really promising results for a project evaluating large text summarisation, so it would be great to understand whether I am using correctly, or if not, why the results are consistent anyway.

**Comment by cyr19** [comment]
Hi. Thanks for your interest.

Yes. It’s ok to provide an empty list for src or any other list for src. As long as you don’t initialize the metric with src or cross_lingual, the list for src will never be used.

Did you also set the NLI weight to 1.0?


Best,

在 2024年2月14日，10:11，WoolnerWoofieWool ***@***.***> 写道：

﻿

I am getting some really promising results for a project evaluating large text summarisation, so it would be great to understand whether I am using correctly, or if not, why the results are consistent anyway.

**Comment by WoolnerWoofieWool** [comment]
Yes. I changed the weight guessing that was why I was getting a max score of about 0.2 for identical texts. Another thing I wasnt clear on was whether NLI was suitable for larger texts. From my calcs it is looking strong but I'm not sure whether it is using all the input for its calcs. Its also surprisingly fast!

**Comment by WoolnerWoofieWool** [comment]
what is srcs for? Is that metadata for the hypotheses?

**Comment by cyr19** [comment]
> what is srcs for? Is that metadata for the hypotheses?

So basically you can evaluate your system summaries either in ref-based env (hypothesis vs. reference summary) or in ref-free env (hypothesis vs. source text to be summarized). If you initialize the metric setting `src' to True, the metric will compare the hypothesis against the source text (srcs) instead of the reference summary (refs) to get the score.
